src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_BEGIN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+import seedu.address.logic.commands.SearchCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+/**
+ * Parses input arguments and creates a new SearchCommand object.
+ */
+public class SearchCommandParser implements Parser<SearchCommand> {
+    /**
+     * Parses the given {@code String} of arguments
+     * and returns a SearchCommand object for execution.
+     *
+     * @param args The input arguments for the search command.
+     * @return A SearchCommand instance containing the parsed begin and end times.
+     * @throws ParseException If the user input does not conform to the expected format.
+     */
+    public SearchCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_BEGIN, PREFIX_END);
+        if (!checkValidPrefixCount(argMultimap) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
```

Comment:
I like the use of constants to make it clear what the output String is representing.